client/rpc/rpcclient_test.go

```diff
@@ -0,0 +1,123 @@
+package rpcclient
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+func GetTestRPCClient() (*RPCClient, error) {
+	connCfg := &rpcclient.ConnConfig{
+		Host:         "localhost:18443", // regtest
+		User:         "test",
+		Pass:         "test",
+		HTTPPostMode: true, // Bitcoin core only supports HTTP POST mode
+		DisableTLS:   true, // Bitcoin core does not provide TLS by default
+	}
+	client, err := New(connCfg, nil)
+	return client, err
+}
+
+type RPCClientTestSuite struct {
+	suite.Suite
+	Client      *RPCClient
+	WalletName  string
+	Address     btcutil.Address
+	BlockCount  int64
+	BlockHashes []*chainhash.Hash
+}
+
+func (suite *RPCClientTestSuite) SetupSuite() {
+	// Initialize client
+	client, err := GetTestRPCClient()
+	assert.NoError(suite.T(), err)
+	suite.Client = client
+
+	// Create wallet
+	suite.WalletName = "testwallet"
+	walletReq := client.CreateWalletAsync(suite.WalletName)
+	client.Send()
+	_, err = walletReq.Receive()
+	assert.NoError(suite.T(), err)
+
+	// Get new address
+	addressReq := client.GetNewAddressAsync(suite.WalletName)
+	client.Send()
+	suite.Address, err = addressReq.Receive()
+	assert.NoError(suite.T(), err)
+
+	// Generate blocks
+	var nBlocks int64 = 20
+	generateReq := client.GenerateToAddressAsync(nBlocks, suite.Address, nil)
+	client.Send()
+	hashes, err := generateReq.Receive()
+	assert.NoError(suite.T(), err)
+	assert.EqualValues(suite.T(), nBlocks, len(hashes))
+	suite.BlockCount = nBlocks
+
+	// Set block hashes
+	genBlockHashReq := client.GetBlockHashAsync(0)
+	client.Send()
+	genBlockHash, err := genBlockHashReq.Receive()
+	assert.NoError(suite.T(), err)
+	suite.BlockHashes = append([]*chainhash.Hash{genBlockHash}, hashes...)
+
+}
+
+func (suite *RPCClientTestSuite) TestSanityBlockCount() {
+
+	client := suite.Client
+	blockCountReq := client.GetBlockCountAsync()
+	client.Send()
+	blockCount, err := blockCountReq.Receive()
+	assert.NoError(suite.T(), err)
+	assert.EqualValues(suite.T(), suite.BlockCount, blockCount)
+}
+func (suite *RPCClientTestSuite) TestGetHashesByRange() {
+
+	client := suite.Client
+	var nBlocks int64 = 10
+	hashes, err := client.getBlockHashesByRange(suite.BlockCount-nBlocks+1, suite.BlockCount)
+	assert.NoError(suite.T(), err)
+	assert.EqualValues(suite.T(), nBlocks, len(hashes))
+	assert.ElementsMatch(suite.T(), suite.BlockHashes[suite.BlockCount-nBlocks+1:suite.BlockCount+1], hashes)
+
+}
+
+func (suite *RPCClientTestSuite) TestGetBlocksByRange() {
+
+	client := suite.Client
+	var nBlocks int64 = 10
+	blocks, err := client.GetBlocksByRange(suite.BlockCount-nBlocks+1, suite.BlockCount)
+	assert.NoError(suite.T(), err)
+	assert.EqualValues(suite.T(), nBlocks, len(blocks))
+	// How to test values?
```

Comment:
Yeah since we do not generate the blocks it is hard to test them thoroughly, but honestly I think it is fine if we just test the blockhashes returned in the list because that is really all we need to test since that's the only new functionality were adding - batching GetBlocks calls. Also I suggest you consider better test coverage through test tables like the format below 
```suggestion
	type args struct {
		minBlockNumber int64
		maxBlockNumber int64
	}
	tests := []struct {
		name    string
		args    args
		want    []*chainhash.Hash
		wantErr bool
	}{
		{
			name: "genesis",
			args: ...,
			want: []*chainhash.Hash{...},
		},
		{
			name: "first_and_second_blocks",
			args: ...,
			want: []*chainhash.Hash{...},
		},
		{
			name: "negative_range",
			args: ...,
			want:    nil,
			wantErr: true,
		},
		{
			name: "future_range",
			args: ...,
			want:    nil,
			wantErr: true,
		},
	}
	for _, tt := range tests {
		s.Run(tt.name, func() {
			...
		})
	}
```

Reply:
Wow this is amazing

Reply:
And that makes sense to just test blockhashes